libraries/AP_HAL_ChibiOS/hwdef/ARK_FPV/hwdef.dat

```diff
@@ -0,0 +1,215 @@
+
+# Use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+define HAL_STORAGE_SIZE 32768
+STORAGE_FLASH_PAGE 14
+
+# To be compatible with the px4 bootloader we need
+# to use a different RAM_MAP
+env USE_ALT_RAM_MAP 1
+
+SERIAL_ORDER OTG1 UART7 UART5 USART1 USART2 UART4 USART6 OTG2
+
+# Debug console
+STDOUT_SERIAL SD3
+STDOUT_BAUDRATE 57600
+
+# Default to all pins low to avoid ESD issues
+DEFAULTGPIO OUTPUT LOW PULLDOWN
+
+# Telem1 - Telem
+PE8  UART7_TX  UART7
+PF6  UART7_RX  UART7
+PF8  UART7_RTS UART7
+PE10 UART7_CTS UART7
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_MAVLink2
+define DEFAULT_SERIAL1_BAUD 115
+
+# Telem2 - VTX (DJI Air Unit)
+PC12 UART5_TX UART5
+PD2  UART5_RX UART5
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_MSP_DisplayPort
+define DEFAULT_SERIAL2_BAUD 115
+
+# GPS1
+PB6 USART1_TX USART1
+PB7 USART1_RX USART1
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_GPS
+
+# Telem3 - VTX (DJI Air Unit, RX only)
+PA3 USART2_RX USART2
+PD5 USART2_TX USART2
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_None # TODO: DJI HDL?
+
+# ESC telem RX only
+PH13 UART4_TX UART4 NODMA
+PH14 UART4_RX UART4
+define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_ESCTelemetry
+define DEFAULT_SERIAL5_BAUD 115
+define HAL_SERIAL_ESC_COMM_ENABLED 1
+define HAL_WITH_ESC_TELEM 1
+
+# RC Input
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_RCIN
+
+# TODO: build was failing
+# define DEFAULT_SERIAL7_PROTOCOL SerialProtocol_SLCAN
+
+# ADCs
+PA0 SCALED1_V3V3 ADC1 SCALE(2)
+PB1 VDD_5V_SENS ADC1 SCALE(2)
+# TODO: 12V monitor
+# PA4 SCALED2_V3V3 ADC1 SCALE(7.66)
+
+PB0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC2 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+# Analog battery input scaling
+# See STM32H743xx.py
+define HAL_BATT_VOLT_PIN 9
+define HAL_BATT_CURR_PIN 12
+define HAL_BATT_VOLT_SCALE 21
+define HAL_BATT_CURR_SCALE 120
+define HAL_BATT_MONITOR_DEFAULT 4
+
+# SPI1 - IIM42653 IMU
+PA5 SPI1_SCK SPI1
+PB5 SPI1_MOSI SPI1
+PG9 SPI1_MISO SPI1
+PI9 IMU1_CS CS
+PF2 IMU1_DRDY INPUT
+
+# SPI6 - external (SPI OSD or IMU)
+PB3  SPI6_SCK SPI6
+PA6  SPI6_MISO SPI6
+PG14 SPI6_MOSI SPI6
+PI10 EXT1_CS CS
+PD11 DRDY_ADIS16507 INPUT GPIO(93)
+
+# GPIO(93) for data ready on ADIS16507
+# define ADIS_DRDY_PIN 93
+
+# Motors, see STM32H743xx.py
+PI0  TIM5_CH4 TIM5 PWM(1) GPIO(50)
+PH12 TIM5_CH3 TIM5 PWM(2) GPIO(51) BIDIR
+PH11 TIM5_CH2 TIM5 PWM(3) GPIO(52)
+PH10 TIM5_CH1 TIM5 PWM(4) GPIO(53) BIDIR
+PI5  TIM8_CH1 TIM8 PWM(5) GPIO(54) BIDIR
+PI6  TIM8_CH2 TIM8 PWM(6) GPIO(55)
+PI7  TIM8_CH3 TIM8 PWM(7) GPIO(56) BIDIR
+PI2  TIM8_CH4 TIM8 PWM(8) GPIO(57)
+PD12 TIM4_CH1 TIM4 PWM(9) GPIO(58)
+
+# PWM output for buzzer
+PF9 TIM14_CH1 TIM14 GPIO(77) ALARM
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+# I2C1 (GPS1 compass, external)
+PB9 I2C1_SDA I2C1
+PB8 I2C1_SCL I2C1
+
+# I2C2 (BMP390 baro, internal)
+PF1 I2C2_SCL I2C2
+PF0 I2C2_SDA I2C2
+
+# I2C4 (IIS2MDC compass, internal)
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
```

Comment:
Both of these busses are "internal". I noticed most other boards only have 1 internal bus, is this okay to have 2 internal busses?  I assume so since the I2C_INTERNAL_MASK bitmask exists, but wanted to double check.

Reply:
yes